server/src/main/java/org/elasticsearch/action/bulk/Retry.java

```diff
@@ -40,12 +40,12 @@
  * Encapsulates synchronous and asynchronous retry logic.
  */
 public class Retry {
-    private final Class<? extends Throwable> retryOnThrowable;
+    private final RestStatus retryOnStatus;
     private final BackoffPolicy backoffPolicy;
     private final Scheduler scheduler;
 
-    public Retry(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy, Scheduler scheduler) {
-        this.retryOnThrowable = retryOnThrowable;
+    public Retry(RestStatus retryOnStatus, BackoffPolicy backoffPolicy, Scheduler scheduler) {
+        this.retryOnStatus = retryOnStatus;
```

Comment:
shall we consider making this hardcoded rather than an argument given that we always pass in the same value for it?

Reply:
Yes, as it's only used here in bulk for this case.